### DISCUSSION TITLE
Question: Dynamic Content Items

### DISCUSSION THREAD

**Discussion by LMSSonos** [OrchardCMS/OrchardCore#15852] [Q&A]
To show the real example, I have following menu structure 
![Unbenannt](https://user-images.githubusercontent.com/19758451/69416953-0f94af00-0d18-11ea-9fa8-4a28ae14fda9.png)

For this I created a taxonomy as follows:
- Winter (url: /winter)
  - Freeride (url: /winter/freeride)
  - Avalanche Training
  - Snowshoing
- Summer
- Outdoor

Then I create ContentItems (Type: ActivityPage) for Freeride Tour1, FreeRide Tour 2, Basic, etc. They all reference the corresponding taxonomy.

Now when I click winter in the menu, I want to show a page which lists all the sub-taxonomies as shown here (call this MainOverview):
![grafik](https://user-images.githubusercontent.com/19758451/69416841-d0665e00-0d17-11ea-9f82-304541c983e8.png)

Then, when selecting the Freeride example, I want to show all Content-Items with that Taxonomy, in this case (Call this SubOverview)

![grafik](https://user-images.githubusercontent.com/19758451/69417028-318e3180-0d18-11ea-9983-6293807ffdcf.png)

**Question**
I could create a MainOverview Page for Winter, Summer, Outdoor and on each Page I select the taxonomy to show.  

Then I could create SubOverview Pages where I assign the subtaxonomy (Freeride) and it shows that ActivityPages with that taxonomy. 

The Problem: With that is, that if the structure is big, you need to create lots of pages. 

I was wondering if there is something more dynamic to build this structure without having to generate a MainOverview Page and SuboverviewPage for each taxonomy.

**Comment by dodyg** [comment]
You can query from TaxonomyIndex 

https://github.com/OrchardCMS/OrchardCore/issues/4494#issuecomment-553877942

So essentially you just need to make two pages: `MainOverview` and `SubOverview` and create a query that take the taxonomy term id

**Comment by LMSSonos** [comment]
@dodyg thanks for your hint. But how can I have like a dynamic content item (page) which will be called if you call /winter o /winter/freeride. I'm looking for a way not having to create all these MainOverview and SubOverview pages.

ps: do you think taxonomy is the right thing here or should I go for lists (like blog template)

**Comment by deanmarcussen** [comment]
Something you are probably looking for is this https://github.com/OrchardCMS/OrchardCore/issues/2688

**Comment by LMSSonos** [comment]
@deanmarcussen thanks! that's exact what I was looking for. As it sounds, it will take still longer until this is released, so for the moment I will probably go the way to build all the pages.

**Comment by dodyg** [comment]
Ah I got it - the problem you were trying to solve is preserve the nice url structure (/winter/free-ride).

If you don't care about this, then you can create a Liquid page that handles these structures via query string e.g. `trip?q=/winter/freeride`.

> ps: do you think taxonomy is the right thing here or should I go for lists (like blog template)

I would choose Taxonomy if I had your scenario.

If you are going to create these pages manually for the routing purposes, I hope you are creating a template and call this template from the pages.

**Comment by LMSSonos** [comment]
@dodyg the liquid page is not an option, as I want to have the nice url ;-) 

> If you are going to create these pages manually for the routing purposes, I hope you are creating a template and call this template from the pages.

I will create a ContentType `SubOverview` and this one will have the current taxonomy. Then I create a entry with route `winter/freeride` , another with route `winter/avalanche-training` etc. Then I also planned to add a templated `Content__Suboverview` where I probably use a query to get all the entries from the database. Or did you suggest it differently?

**Comment by dodyg** [comment]
Yeah that should work.

The other approach is to create a custom module to render taxonomy pages in case they grow too large.

**Comment by LMSSonos** [comment]
> The other approach is to create a custom module to render taxonomy pages in case they grow too large.

I tried to use modules (https://orchardcore.readthedocs.io/en/dev/docs/reference/core/Modules/) but the issue I had was that the html code I return from there renders as whole page, but I want to write from the module only to the `{% render_body %}` of part within template. Then this would be the best way. I did not find a way to do this.
